Kinvey.Core/Store/ReadRequest.cs

```diff
@@ -310,7 +325,27 @@ private string BuildIDsQuery(List<string> listIDs)
 			return query.ToString();
 		}
 
-		protected class NetworkReadResponse<T>
+        private async Task<NetworkReadResponse<T>> PerformNetworkGet(string mongoQuery)
+        {
+            var results = await RetrieveNetworkResults(mongoQuery);
+            Cache.Clear(Query?.Expression);
+            Cache.RefreshCache(results);
+            return new NetworkReadResponse<T>(results, results.Count, false);
+        }
+
+        private async Task<NetworkReadResponse<T>> PerformNetworkInitialDeltaGet(string mongoQuery)
+        {
+            var getResult = Client.NetworkFactory.buildGetRequest<T>(Collection, mongoQuery);
+            List<T> results = await getResult.ExecuteAsync();
+            Cache.Clear(Query?.Expression);
+            Cache.RefreshCache(results);
+            string lastRequestTime = getResult.RequestStartTime;
+            var qci = new QueryCacheItem(Collection, mongoQuery, lastRequestTime);
```

Comment:
@vinaygahlawat don't you need to check if the query already exists in the local cache?

Reply:
In this case, it is known that this is the initial GET. The check for the query happens earlier:

https://github.com/Kinvey/dotnet-sdk/blob/feature/MLIBZ-2390-Server_side_delta_set/Kinvey.Core/Store/ReadRequest.cs#L218